config/config.go

```diff
@@ -125,20 +128,55 @@ func HasConfigurationFile(fs afero.Fs, file string) bool {
 	return !os.IsNotExist(err)
 }
 
-func NewConfig(b []byte) (*Config, error) {
-	conf := DefaultConfig()
+func VerifyConfig(fs afero.Fs, config *Config) error {
+	// TODO more validiation?
+	// TODO make directory if it doesn't exist?
+	// Check that config.Keydir exists
+	_, err := fs.Stat(config.Keydir)
+	if os.IsNotExist(err) {
+		return err
+	}
+
+	return nil
+}
+
+func NewConfig(fs afero.Fs, b []byte) (*Config, error) {
+	conf := &Config{}
 	if err := yaml.Unmarshal(b, conf); err != nil {
 		return nil, err
 	}
-	return conf, nil
+	var newConf = new(Config)
+	if conf.Datadir != "" {
+		newConf = MkConfig(conf.Datadir)
+	} else {
+		newConf = DefaultConfig()
+	}
+
+	if err := yaml.Unmarshal(b, newConf); err != nil {
+		return nil, err
+	}
+
+	// TODO
+	// Should sbctl make the directories if they do not exist?
+	if err := VerifyConfig(fs, newConf); err != nil {
+		return nil, fmt.Errorf("invalid sbctl config; %w", err)
+	}
```

Comment:
This doesn't belong in this PR. So lets remove it and just focus on the yubikey part.